py-powsybl/README.md

```diff
@@ -27,6 +27,46 @@ Start the py-powsybl 'server' in a console.
 Demo1.py demostrates what can be done (it executes a loadflow on a network, opens a switch, exports a network to a file in xiidm format)
 
     python Demo1.py     
+    
+Another example using pypowsybl module
+    
+```
+from pypowsybl import *
+
+# simple dump network flows function
+def dump_lines_flow(network):
+    print(len(network.get_lines()))
+    for l in network.get_lines():
+        print(l.get_id() + ";" + str(l.get_terminal_1().get_i()) + ";" + str(l.get_terminal_2().get_i()))
+        print(l.get_current_limits_1().get_permanent_limit())
+        print(str(l.check_permanent_limit_1()) + ";" + str(l.check_permanent_limit_2()))
+        print(str(l.check_permanent_limit_1(0.1)) + ";" + str(l.check_permanent_limit_2(0.1)))
+        print(str(l.is_overloaded()) + ";" + str(l.is_overloaded(0.1)))
+
+if __name__ == '__main__':
+    port = 3338
+    launch("config2", port)
```

Comment:
Would it be possible, instead of launch + shutdown, to be able to use the "with" keyword?
```
with launch("config2", port):
     //do something
//automatic shutdown.
```
It's equivalent to the "try with resource" in java. It is implemented by having an object with __enter__ and __exit__ method I think.

Reply:
use contextlib https://docs.python.org/3/library/contextlib.html